csharp/src/Bargreen.Services/InventoryService.cs

```diff
@@ -101,13 +108,67 @@ public IEnumerable<AccountingBalance> GetAccountingBalances()
                      ItemNumber = "fbr77",
                      TotalInventoryValue = 17.99M
                 }
-            };
+                };
+            });
         }
 
-        public static IEnumerable<InventoryReconciliationResult> ReconcileInventoryToAccounting(IEnumerable<InventoryBalance> inventoryBalances, IEnumerable<AccountingBalance> accountingBalances)
+        public async Task<IEnumerable<InventoryReconciliationResult>> ReconcileInventoryToAccounting(IEnumerable<InventoryBalance> inventoryBalances, IEnumerable<AccountingBalance> accountingBalances)
         {
             //TODO-CHALLENGE: Compare inventory balances to accounting balances and find differences
-            throw new NotImplementedException();
+
+            //insntantiate list of InventoryRecResults
+            List<InventoryReconciliationResult> inventoryReconciliationResults = new List<InventoryReconciliationResult>();
+           
+            // iterate through both the inventtory and accounting balances
+            foreach(InventoryBalance InvItem in inventoryBalances)

```

Comment:
This nested loop will result in algorithmic complexity of O(N^2)...  Not a problem for our small sample size of ~10 items, but problematic if the system were to scale. Can you think of a way to do the reconciliation with O(N) (i.e. looping through each array only once)?